src/mono/mono/metadata/marshal-ilgen.c

```diff
@@ -6193,8 +6267,47 @@ emit_marshal_variant_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 	return conv_arg;
 }
 
+static gboolean
+emit_managed_wrapper_validate_signature (MonoMethodSignature* sig, MonoMarshalSpec** mspecs, MonoError* error)
+{
+	if (mspecs && error) {
```

Comment:
I don't like using `MonoError` as a flag like this. Let's keep `MonoError` usage predictable and boring in the entire runtime.